### PR TITLE
DateTimePicker from DS used in Strapi

### PR DESCRIPTION
**Pull request by simotae14**
What does it do?
We want to integrate the new DateTimePicker inside the design system into the Strapi project.

Decision
I think there is no reason to maintain the DateTimePicker inside the helper-plugin because it is a Design Component that can be used in other places. 

Consequences
I think we can have a more clear and correct division between components into the different repos. We need also to take care about the possible errors with the old versions of the Design System and strapi

Display a warning to the user that inside the old DateTimePicker that this component is still under definition/development and that it is not recommended to use it because the code and the API will change soon.

How to test it?
We need to test if we make some regressions in the code, in every section where we can use the DateTimePicker, how it works with content from another locale

Why is it needed?
We need to move the logic from the helper-plugin to the Design System project

### CODE REVIEW

File: packages/core/admin/package.json
Comment (joshuaellis) [inline packages/core/admin/package.json:55]: I'm assuming if all these tests pass we can release the stable and use that instead?

**Review by joshuaellis — CHANGES_REQUESTED** [review]

File: packages/core/admin/package.json
Comment (simotae14) [reply packages/core/admin/package.json:55]: If the integration with beta works yes, we can release a stable version of the DS and just use it inside this branch and then release this branch

File: packages/core/helper-plugin/package.json
Comment (gu-stav) [inline packages/core/helper-plugin/package.json:73]: Please make sure you update this dependency too in both locations :)

**Review by joshuaellis — DISMISSED** [review]

File: packages/core/helper-plugin/package.json
Comment (gu-stav) [inline packages/core/helper-plugin/package.json:73]: Just to make sure because @joshuaellis has approved it already: do we want to have a beta release in the `main` branch or should we wait for a final `1.3.0` release?

**Review by gu-stav — CHANGES_REQUESTED** [review]

File: packages/core/helper-plugin/package.json
Comment (joshuaellis) [reply packages/core/helper-plugin/package.json:73]: I believe the plan is (or should be) when it's been QA'd & ready to merge from all aspects, to release 1.3.0 stable and include it here before merging to main. @simotae14 is gonna talk to me when he's ready for that, if i've understood everything correctly.

Comment (gu-stav) [reply packages/core/helper-plugin/package.json:73]: Ok. I'll add a no merge label instead. Thanks!

Comment (simotae14) [reply packages/core/helper-plugin/package.json]: yes, now I have updated with the version 1.4.0

File: packages/core/helper-plugin/lib/src/components/FilterPopoverURLQuery/Inputs.js
Comment (joshuaellis) [inline packages/core/helper-plugin/lib/src/components/FilterPopoverURLQuery/Inputs.js:7]: These can all be imported from the root
```suggestion
import { Field, FieldInput } from '@strapi/design-system';
```

**Review by joshuaellis — CHANGES_REQUESTED** [review]

File: packages/core/helper-plugin/lib/src/components/FilterPopoverURLQuery/Inputs.js
Comment (simotae14) [reply packages/core/helper-plugin/lib/src/components/FilterPopoverURLQuery/Inputs.js:7]: ok thanks @joshuaellis

**Review by joshuaellis — APPROVED** [review]

File: packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js
Comment (simotae14) [reply packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js]: Can you take a look at my last implementation to check if it I understood your requests, thanks

File: packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js
Comment (joshuaellis) [inline packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js:9]:
```suggestion
```

Don't log this 😅

File: packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js
Comment (joshuaellis) [inline packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js:4]:
```suggestion
// TODO: remove DateTimePicker component from the helper-plugin in V5
export default function DateTimePicker(props) {
```

Instead, write it here.

**Review by joshuaellis — CHANGES_REQUESTED** [review]
I think you slightly misconstrued @gu-stav's feedback

File: packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js
Comment (simotae14) [reply packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js:5]: ok thanks, done

File: packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js
Comment (gu-stav) [inline packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js:10]: Can we delete this file now (I don't see it showing up in the diff)?

File: packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js
Comment (gu-stav) [inline packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js:5]: I think I'd prefer if we'd rename the export (shouldn't be a problem since it is a default export) to `export default function DateTimePickerLegac(props)` and then import/ export the `DateTimePicker` component from the design-system with renaming it.

**Review by joshuaellis — APPROVED** [review]
from my POV this is fine, I can see there's comments from Gustav remaining though :)

File: packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js
Comment (simotae14) [reply packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js:10]: yes removed and also removed the test

File: packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js
Comment (simotae14) [reply packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js:5]: Yes that's ok for me, it is something that we need to have just until v5 so it is ok to name it DateTimePickerLegacy

**Review by joshuaellis — APPROVED** [review]

File: packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js
Comment (gu-stav) [reply packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js]: Makes sense, thanks. Can you please assign the `selectButtonTitle` prop using `formatMessage`? This allows users to overwrite value for different admin locales?

File: packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js
Comment (gu-stav) [reply packages/core/helper-plugin/lib/src/components/DateTimePicker/index.js]: Thanks!

**Review by gu-stav — APPROVED** [review]
Thank you 🌻 Just to make sure: I think we should not merge this before today's release and do it afterwards, to give it ~2 weeks on the main branch to be tested. Do you agree?

**Comment by gu-stav** [comment]
@simotae14 Could you please fix the merge conflicts and merge this PR? 🙏🏼

**Comment by simotae14** [comment]
yes sure

**Pull request merged**
